src/integrations/BingAds/utils.js

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable camelcase */
+const EXCLUSION_KEYS = [
+  'event',
+  'category',
+  'category_id',
+  'currency',
+  'total',
+  'value',
+  'revenue',
+  'ecomm_category',
+  'transaction_id',
+  'order_id',
+  'checkout_id',
+  'ecomm_pagetype',
+  'pagetype',
+  'query',
+  'products',
+  'product_id',
+  'sku',
+  'event_action',
+];
+
+const DEFAULT_PAGETYPE = 'other';
+
+const buildCommonPayload = (message) => {
+  const { event, properties = {} } = message;
+  const { category, currency, total, value, revenue } = properties;
+  const payload = {
+    event_label: event,
+    event_category: category,
+    currency,
+    revenue_value: total || value || revenue,
+  };
+  return payload;
+};
+
+const handleProductsArray = (properties) => {
+  const productIds = [];
+  const items = [];
+
+  const products = Array.isArray(properties.products) ? properties.products : [properties];
+
+  products.forEach((product) => {
+    const { product_id, sku, price, quantity = 1 } = product;
+    const productId = product_id || sku;
```

Comment:
Any reason we are keeping `sku` here as well?

Reply:
keeping sku as fallback valid If product_id is not there. Doing there for all ecomm integrations.